<?xml version="1.0"?>
<launch>
    <arg name="sensor_model" description="Pandar64|Pandar40P|PandarXT16|PandarXT32|PandarXT32M|PandarAT128|PandarQT64"/>
    <arg name="return_mode" default="Dual" description="See readme for supported return modes"/>
    <arg name="frame_id"  default="hesai"/>
    <arg name="scan_phase" default="0.0" />

    <arg name="calibration_file" default="$(find-pkg-share nebula_decoders)/calibration/hesai/$(var sensor_model).csv"/>
    <arg name="correction_file" default="$(find-pkg-share nebula_decoders)/calibration/hesai/$(var sensor_model).dat"/>

    <arg name="bag_path" default="path to bag dir"/>
    <arg name="storage_id" default="sqlite3"/>
    <arg name="out_path" default="path to output dir"/>
    <arg name="format" default="cdr"/>
    <arg name="target_topic" default="/pandar_packets"/>


    <node pkg="nebula_offline_sample" exec="hesai_ros_offline_extract_pcd_node"
          name="hesai_cloud" output="screen">
        <param name="sensor_model" value="$(var sensor_model)"/>
        <param name="return_mode" value="$(var return_mode)"/>
        <param name="frame_id" value="$(var frame_id)"/>
        <param name="scan_phase" value="$(var scan_phase)"/>
        <param name="calibration_file" value="$(var calibration_file)"/>
        <param name="correction_file" value="$(var correction_file)"/>
        <param name="bag_path" value="$(var bag_path)"/>
        <param name="storage_id" value="$(var storage_id)"/>
        <param name="out_path" value="$(var out_path)"/>
        <param name="format" value="$(var format)"/>
        <param name="target_topic" value="$(var target_topic)"/>
    </node>
</launch>
